src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-configuration-page/rgw-configuration-page.component.ts

```diff
@@ -136,19 +137,24 @@ export class RgwConfigurationPageComponent extends ListWithDetails implements On
     super.setExpandedRow(expandedRow);
   }
 
+  flattenData(data: any) {
+    const combinedArray: any[] = [];
+    for (const kmsData of Object.values(data.kms)) {
+      combinedArray.push(kmsData);
+    }
+    for (const s3Data of Object.values(data.s3)) {
+      combinedArray.push(s3Data);
+    }
+    return combinedArray;
+  }
+
   fetchData() {
     this.rgwBucketService.getEncryptionConfig().subscribe((data: any) => {
       this.allEncryptionValues = data;
-      const allowedBackends = rgwBucketEncryptionModel.kmsProviders;
-
-      const kmsBackends = this.getBackend(data, 'SSE_KMS');
-      const s3Backends = this.getBackend(data, 'SSE_S3');
-
-      const allKmsBackendsPresent = this.areAllAllowedBackendsPresent(allowedBackends, kmsBackends);
-      const allS3BackendsPresent = this.areAllAllowedBackendsPresent(allowedBackends, s3Backends);
-
-      this.disableCreate = allKmsBackendsPresent && allS3BackendsPresent;
-      this.encryptionConfigValues = Object.values(data).flat();
+      const kmsBackends = this.getBackend(data, 'kms');
+      const s3Backends = this.getBackend(data, 's3');
```

Comment:
You should be passing here enum encryption type.
Then also use that enum in types in function call and interface def